modules/services/model/src/main/java/au/com/shiftyjelly/pocketcasts/models/db/dao/BookmarkDao.kt

```diff
@@ -94,6 +95,36 @@ abstract class BookmarkDao {
     )
     abstract fun findBookmarksFlow(deleted: Boolean = false): Flow<List<Bookmark>>
 
+    @Query(
+        """SELECT *
+            FROM bookmarks
+            WHERE deleted = :deleted
+            ORDER BY 
+            CASE WHEN :isAsc = 1 THEN created_at END ASC, 
+            CASE WHEN :isAsc = 0 THEN created_at END DESC""",
+    )
+    abstract fun findAllBookmarksOrderByCreatedAtFlow(
+        isAsc: Boolean,
+        deleted: Boolean = false,
+    ): Flow<List<Bookmark>>
+
+    // Sorts the array by podcast name, episodes release date, and the bookmarks timestamp
+    @Query(
+        """SELECT bookmarks.*, podcasts.title as podcastTitle, podcast_episodes.title as episodeTitle, podcast_episodes.published_date as publishedDate
+            FROM bookmarks
+            LEFT JOIN podcast_episodes ON bookmarks.episode_uuid = podcast_episodes.uuid
+            LEFT JOIN podcasts ON bookmarks.podcast_uuid = podcasts.uuid
+            WHERE deleted = :deleted
+            ORDER BY
+            CASE WHEN podcasts.title is NULL then 1 else 0 end, UPPER(podcasts.title) ASC, 
```

Comment:
It'd be nice to escape English articles, similarly to what we do here.

https://github.com/Automattic/pocket-casts-android/blob/c9127eb0bfe5c76ea157697904d1fdbcfd64f1dd/modules/services/model/src/main/java/au/com/shiftyjelly/pocketcasts/models/db/dao/EpisodeDao.kt#L68-L84

Reply:
I can also recommend running the query through a formatter like [this one](https://codebeautify.org/sqlformatter) if it gets too messy. 😅